lib/zendesk_api/resources.rb

```diff
@@ -500,6 +500,14 @@ class SuspendedTicket < ReadResource
     put :recover
   end
 
+  class TargetFailure < ReadResource
+    def target
+      @client.targets.to_a.find do |target|
```

Comment:
If you use `.all` in stead of `to_a` you should be able to reduce the number of calls in case of pagination.